lib/compiler/node/index.ts

```diff
@@ -39,6 +52,10 @@ const instantiateWASM = async (
 };
 
 const startRunningService = async () => {
+  for (const key of Object.keys(sharedFunctions)) {
+    const globalKey = `__astro_${key}`;
+    (globalThis as any)[globalKey] = (sharedFunctions as any)[key];
+  }
```

Comment:
Don't love this but exposing functions on `globalThis` is extremely straightforward. I couldn't seem to pass the function directly because then you end up in `wasm_exec.js` memory allocation hell.